### PR TITLE
 Added ContextualType.EnumerableItemType to inspect nullability of collections

### PR DESCRIPTION
**Pull request by jeremyVignelles**
There are still some issues around List<T?> and more generally `GenericClass<T?>` NRT inference.

### CODE REVIEW

File: src/Namotion.Reflection/TypeExtensions.cs [hunk 1]
```diff
@@ -75,10 +75,11 @@ public static bool InheritsFromTypeName(this Type type, string typeName, TypeNam
         /// <summary>Gets the type of the array item.</summary>
         public static Type? GetEnumerableItemType(this Type type)
         {
-            var elementType = type.GetElementType();
-            if (elementType != null)
+            var genericTypeArguments = type.GetGenericTypeArgumentsOfTypeOrBaseTypes();
```
Comment (jeremyVignelles) [inline src/Namotion.Reflection/TypeExtensions.cs:78]: Why this change, and why is it different from the EnumerableItemType property now ?

Comment (RicoSuter) [reply src/Namotion.Reflection/TypeExtensions.cs:78]: We use generic type arguments also for EnumerableItemType now, so that your tests work... that's why I'd also keep it here... also because it's probably faster and the same as before, no?

Comment (RicoSuter) [reply src/Namotion.Reflection/TypeExtensions.cs:78]: In theory we should probably check in both versions whether there is a GetEnumerator method?

Comment (jeremyVignelles) [reply src/Namotion.Reflection/TypeExtensions.cs:78]: Don't understand your last question. Do you want to chat on gitter?
